sdk_lib/sdk_container_common.sh

```diff
@@ -0,0 +1,104 @@
+#
+# This file contains common functions used across SDK container scripts.
+
+#
+# globals
+#
+sdk_container_common_versionfile="sdk_container/.repo/manifests/version.txt"
+
+# Common "echo" function
+
+function yell() {
+    echo -e "\n###### $@ ######"
+}
+# --
+
+# Prevent the sourcetree from being used by multiple containers with different
+#  SDK release versions by use of a version lock.
+# The lock is self-cleaning and does not need to be removed in scripts calling
+# this function.
+function lock_version() {
+    local version="$1"
+    local owner="$2"
+    local lock_version=""
+    local lock_owner=""
+    local lock_owner_active=""
+
+    if ! mkdir sdk_container/.lock 2>/dev/null; then
+
+        if [ -f sdk_container/.lock/version ] ; then
+            lock_version="$(cat sdk_container/.lock/version)"
+        fi
+        if [ -f sdk_container/.lock/owner ] ; then
+            lock_owner="$(cat sdk_container/.lock/owner)"
+            lock_owner_active="$(docker ps --no-trunc --filter name="$lock_owner" --format '{{.Status}}')"
+        fi
+
+        if [ -z "$lock_owner_active" ] ; then
+            yell "Removing stale SDK lock (owner: '$lock_owner', version: '$lock_version')"
+            rm -f sdk_container/.lock/version sdk_container/.lock/owner
+        elif [ "$lock_version" != "$version" ] ; then
+            echo "ERROR: container workspace is locked." >&2
+            echo "This means that an active SDK container is using this workspace."  >&2
+            echo "(owner: $lock_owner, version: $lock_version)"  >&2
+            return 1
+        fi
+    fi
+
+    echo "$version" > sdk_container/.lock/version
+    echo "$owner" > sdk_container/.lock/owner
+}
+# --
+
+# Creates the Flatcar build / SDK version file. We assume that both
+#  Flatcar OS version and SDK version are the same.
+# Must be called from the script root.
+
+function create_versionfile() {
+    local version="$1"
+
+    yell "Writing versionfile '$sdk_container_common_versionfile', setting version '$version'."
+
+	cat >"$sdk_container_common_versionfile" <<EOF
+FLATCAR_VERSION=$version
+FLATCAR_VERSION_ID=$version
+FLATCAR_BUILD_ID=""
+FLATCAR_SDK_VERSION=$version
+EOF
+}
+# --
+
+
+function get_version_from_versionfile() {
+    if [ -f "$sdk_container_common_versionfile" ]; then
+        echo "$(source "$sdk_container_common_versionfile"; echo "$FLATCAR_SDK_VERSION")"
+    fi
+}
+# --
+
+
+# Check out ebuild repos tag/branch/commit-ish.
+# Must be called from the script root.
+
+function checkout_ebuild_repos() {
+    local version="$1"
+
+    yell "Checking out portage-stable, coreos-overlay tags"
+    for repo in portage-stable coreos-overlay; do
+		cd "sdk_container/src/third_party/$repo"
+		git fetch --all
+
+		tag=$(git tag -l | grep "${version}" || true)
+		if [ -z "$tag" ]; then
+			echo "ERROR: Unable to find tag for version '$version' in repo '$repo'"
+			exit 1
+		fi
+
+	    yell "$repo: $tag"
+		git checkout "$tag"
```

Comment:
I think this should use the submodule commit. Would it work to make a symlink to the submodule checkout instead of checking out another version?

Reply:
Hmmm, good point. We would need to check out the respective `flatcar-scripts` tag, then update the submodules.